api/src/identity-access-management/infrastructure/repositories/authentication-method.repository.js

```diff
@@ -269,6 +270,33 @@ const batchUpdatePasswordThatShouldBeChanged = function ({
   );
 };
 
+/**
+ * @param {array} userIds
+ * @param {number} chunkSize
+ * @returns {Promise<{linesUpdated: number}>}
+ */
+const batchAnonymizeByUserIds = async function ({ userIds, chunkSize = 1000 }) {
+  const userIdBatches = _.chunk(userIds, chunkSize);
+  let anonymizedUserIds = 0;
+
+  await bluebird.mapSeries(userIdBatches, async (userIdBatch) => {
+    const anonymizedUserIdBatch = await knex(AUTHENTICATION_METHODS_TABLE)
+      .whereIn('userId', userIdBatch)
+      .andWhere('identityProvider', 'GAR')
+      .update(
+        {
+          authenticationComplement: { firstName: 'anonymized', lastName: 'anonymized' },
+          updatedAt: knex.fn.now(),
+          externalIdentifier: knex.raw('CONCAT(\'anonymized-\', "authentication-methods".id)'),
+        },
+        [COLUMNS],
+      );
+    anonymizedUserIds += anonymizedUserIdBatch.length;
+  });
```

Comment:
Et voici une implémentation sans `bluebird` de manière à ne pas introduire de nouveaux usages de `bluebird` sachant que cette bibliothèque est en fin de vie : 

```suggestion
  for (const userIdBatch of userIdBatches) {
    const anonymizedUserIdBatch = await knex(AUTHENTICATION_METHODS_TABLE)
      .whereIn('userId', userIdBatch)
      .andWhere('identityProvider', 'GAR')
      .update(
        {
          authenticationComplement: { firstName: 'anonymized', lastName: 'anonymized' },
          updatedAt: knex.fn.now(),
          externalIdentifier: knex.raw('CONCAT(\'anonymized-\', "authentication-methods".id)'),
        },
        [COLUMNS],
      );
    anonymizedUserIds += anonymizedUserIdBatch.length;
  }
```